pymc4/distributions/multivariate.py

```diff
@@ -195,7 +196,11 @@ def __init__(self, name, loc, covariance_matrix, **kwargs):
     @staticmethod
     def _init_distribution(conditions):
         loc, covariance_matrix = conditions["loc"], conditions["covariance_matrix"]
-        return tfd.MultivariateNormalFullCovariance(loc=loc, covariance_matrix=covariance_matrix)
+        try:
+            chol_cov_matrix = tf.linalg.cholesky(covariance_matrix)
+        except tf.errors.InvalidArgumentError:
+            raise ValueError("Cholesky decomposition failed! Check your `covariance_matrix`.")
```

Comment:
Is this guaranteed to be the case in an InvalidArgumentError?

Reply:
It throws ``InvalidArgumentError`` for float datatypes while ``NotNoundError`` for integer and other data types. Do you think we should be handling data types other than floats?